merkle_tree.py

```diff
@@ -12,12 +12,13 @@ def __init__(self, hash):
 class MerkleTree:
     def __init__(self):
         self.leaves = []
-        self.root = None
+        self.root: MerkleNode = None
 
     def add_leaf(self, leaf_hash):
         new_leaf = MerkleNode(leaf_hash)
         self.leaves.append(new_leaf)
-        self.recalculate_tree()
+        # Do we need to calculate the tree everytime we add a leaf?
+        # self.recalculate_tree()
```

Comment:
no, I think when we have 10 txs - only then we can build a tree

Reply:
perfect, I will remove it then